### ISSUE TITLE
I still don't understand how to adjust to get the best video according to the chart

### ISSUE THREAD

**Issue by thanhtamns** [Dorian25/ytb-most-replayed#2]
Hi Mate,

I don't know how the program can retrieve and download the best video in the youtube url. Currently the program is downloading the entire length of the video on YouTube
![image](https://github.com/Dorian25/ytb-most-replayed/assets/14258304/f91ab131-7442-4bfd-bd53-fbcc01850cb0)

**Comment by Dorian25** [comment]
Hi man, first of all thanks for your interest in my program !

The program allows you to extract "the most replayed moments" from a youtube video (from its url).

I don't know if you've noticed, but when you move your mouse over the video's progress bar, you can see a sort of graph displaying the "most replayed" moments.
![image](https://github.com/Dorian25/ytb-most-replayed/assets/32178615/043fc328-9d14-4cbf-9b06-b68b8a1eb7aa)

So my program has to download the whole video to cut out the most replayed moments.

I don't know if that's clearer?

**Comment by thanhtamns** [comment]
Thank you for your reply. But as I understand it, the result of the final video is the best cut from the original video?
But I don't see any short videos being extracted, only videos with the same length as videos downloaded from YouTube exist?

**Comment by Dorian25** [comment]
> But as I understand it, the result of the final video is the best cut from the original video?

Not exactly, as you said my program extracts by default 10 seconds (5 sec before / 5 sec after) from each of the most replayed moments. It's a very good idea to group/merge all the clips into one video 👍 

> But I don't see any short videos being extracted, only videos with the same length as videos downloaded from YouTube exist?

normally they should appear in the `./peaks` folder but I'm going to retest all that...

EDIT : I use the `yt-dlp` library to download the video but it seems that it is not functional at the moment (ERROR 416)

**Comment by thanhtamns** [comment]
I find the cutting software is not accurate. I tested it, in a soccer match, there were 2 tyr numbers of 2-2, but only a single clip of a goal was published.
Error 416 was resolved by fixing this code:
![image](https://github.com/Dorian25/ytb-most-replayed/assets/14258304/1951e8ec-36a0-42ee-a666-5b2d4b52ed1c)

**Comment by Dorian25** [comment]
Thx for debugging ! Can you send me the link to the youtube video ?

**Comment by thanhtamns** [comment]
https://www.youtube.com/watch?v=3j_daNHio4o&ab_channel=FCBarcelona

**Comment by Dorian25** [comment]
Yep , I only see one "most replayed" moment at 1:40:12 ! But You can lower the `threshold` as you wish to capture lowest peaks.

![image](https://github.com/Dorian25/ytb-most-replayed/assets/32178615/e6327fe8-26f1-42d3-8c57-1df204ced8d0)


![image](https://github.com/Dorian25/ytb-most-replayed/assets/32178615/745c32e9-f7e2-4b7f-9cc9-d568c69032f7)

**Comment by Dorian25** [comment]
@thanhtamns What would you have expected as a result with the video you sent me ?

**Comment by thanhtamns** [comment]
If possible, the software will be smarter when automatically recognizing the best moment in the video.
Because if I make a large number of videos, I find it takes a lot of time for the software to process them.
